EIPS/eip-3023.md

```diff
@@ -0,0 +1,193 @@
+---
+eip: 3023
+title: Contract Wallet Detection
+author: Ricardo Guilherme Schmidt (@3esmit)
+discussions-to: https://ethereum-magicians.org/t/informational-3023-contract-wallet-detection/4831
+status: Draft
+type: Informational
+created: 2020-10-04
+requires: 20, 897, 1820
+---
+
+## Simple Summary
+
+Provides a registry for checking if a contract is a user wallet.
+
+## Abstract
+
+The wallet software should warn users of common mistakes, such as transferring ERC20 to a contract which is not intended to handle it. User wallet contracts can be detected off-chain through analysis of common code hashes used for most users. 
+
+## Motivation
+
+Huge amounts of value have been lost due simple mistakes as sending all your tokens to the token contract itself, a common mistake by new (or tired) users, to paste the token address in the destination field, instead of pasting it on the wallet configuration. 
+
+Also, new users, coming from less advanced blockchains where all addresses are controlled by a person, willing to participate in a ethereum project, ends up sending value to the wrong contract, and/or directly (without using ERC20 approval design) to the project's contract.
```

Comment:
I would change to "end up sending" instead of "ends up sending". If you take all the extras away from the sentence it would say "New users end up sending value to the wrong contract" which sounds better than "ends up".

Reply:
Thanks, I did as you suggested.